pl_bolts/models/gans/pix2pix/components.py

```diff
@@ -1,147 +1,129 @@
-    def __init__(self, in_channels, out_channels):
+    def __init__(self, in_channels: int, out_channels: int) -> None:
         """Paper details:
 
         - Encoder: C64-C128-C256-C512-C512-C512-C512-C512
-        - All convolutions are 4×4 spatial filters applied with stride 2
+        - All convolutions are 4x4 spatial filters applied with stride 2
         - Convolutions in the encoder downsample by a factor of 2
-        - Decoder: CD512-CD1024-CD1024-C1024-C1024-C512 -C256-C128
+        - Decoder: CD512-CD1024-CD1024-C1024-C1024-C512-C256-C128
         """
         super().__init__()
 
         # encoder/donwsample convs
-        self.encoders = [
-            DownSampleConv(in_channels, 64, batchnorm=False),  # bs x 64 x 128 x 128
-            DownSampleConv(64, 128),  # bs x 128 x 64 x 64
-            DownSampleConv(128, 256),  # bs x 256 x 32 x 32
-            DownSampleConv(256, 512),  # bs x 512 x 16 x 16
-            DownSampleConv(512, 512),  # bs x 512 x 8 x 8
-            DownSampleConv(512, 512),  # bs x 512 x 4 x 4
-            DownSampleConv(512, 512),  # bs x 512 x 2 x 2
-            DownSampleConv(512, 512, batchnorm=False),  # bs x 512 x 1 x 1
-        ]
+        self.encoders = nn.ModuleList(
+            [
+                DownSampleConv(in_channels, 64, batchnorm=False),  # bs x 64 x 128 x 128
+                DownSampleConv(64, 128),  # bs x 128 x 64 x 64
+                DownSampleConv(128, 256),  # bs x 256 x 32 x 32
+                DownSampleConv(256, 512),  # bs x 512 x 16 x 16
+                DownSampleConv(512, 512),  # bs x 512 x 8 x 8
+                DownSampleConv(512, 512),  # bs x 512 x 4 x 4
+                DownSampleConv(512, 512),  # bs x 512 x 2 x 2
+                DownSampleConv(512, 512, batchnorm=False),  # bs x 512 x 1 x 1
+            ]
+        )
 
         # decoder/upsample convs
-        self.decoders = [
-            UpSampleConv(512, 512, dropout=True),  # bs x 512 x 2 x 2
-            UpSampleConv(1024, 512, dropout=True),  # bs x 512 x 4 x 4
-            UpSampleConv(1024, 512, dropout=True),  # bs x 512 x 8 x 8
-            UpSampleConv(1024, 512),  # bs x 512 x 16 x 16
-            UpSampleConv(1024, 256),  # bs x 256 x 32 x 32
-            UpSampleConv(512, 128),  # bs x 128 x 64 x 64
-            UpSampleConv(256, 64),  # bs x 64 x 128 x 128
-        ]
-        self.decoder_channels = [512, 512, 512, 512, 256, 128, 64]
+        self.decoders = nn.ModuleList(
+            [
+                UpSampleConv(512, 512, dropout=True),  # bs x 512 x 2 x 2
+                UpSampleConv(1024, 512, dropout=True),  # bs x 512 x 4 x 4
+                UpSampleConv(1024, 512, dropout=True),  # bs x 512 x 8 x 8
+                UpSampleConv(1024, 512),  # bs x 512 x 16 x 16
+                UpSampleConv(1024, 256),  # bs x 256 x 32 x 32
+                UpSampleConv(512, 128),  # bs x 128 x 64 x 64
+                UpSampleConv(256, 64),  # bs x 64 x 128 x 128
+            ]
+        )
```

Comment:
these comments regarding shapes are only true in case of an image of the shape 256x256.